### PR TITLE
Use type aliases for CoEnv in MapFunc.

### PR DESCRIPTION
**Pull request by alissapajer**

### CODE REVIEW

File: connector/src/main/scala/quasar/qscript/MapFunc.scala [hunk 1]
```diff
@@ -50,41 +50,50 @@ sealed abstract class Ternary[T[_[_]], A] extends MapFunc[T, A] {
 object MapFunc {
   import MapFuncs._
 
+  type CoMF[T[_[_]], A, B] = CoEnv[A, MapFunc[T, ?], B]
+
+  type TCoMF[T[_[_]], A] = T[CoMF[T, A, ?]]
+  type CoMFR[T[_[_]], A] = CoMF[T, A, TCoMF[T, A]]
```
Comment (paulp) [inline connector/src/main/scala/quasar/qscript/MapFunc.scala]: I am on board with the idea, but these aliases cross some imaginary line in my head where they become too difficult to understand. To me the major benefit of type aliases is common subexpression elimination and limiting the nesting of square bracket expressions. Shortening names is more mixed in its impact. What do you think about these being `CoMapFunc`, `TCoMapFunc`, and `CoMapFuncR` ?

File: connector/src/main/scala/quasar/qscript/MapFunc.scala [hunk 2]
```diff
@@ -50,41 +50,50 @@ sealed abstract class Ternary[T[_[_]], A] extends MapFunc[T, A] {
...
+
+  private def makeCoMF[T[_[_]], A, B](mf: A \/ MapFunc[T, B]): CoMF[T, A, B] =
+    CoEnv[A, MapFunc[T, ?], B](mf)
+
+  private def makeCoMFR[T[_[_]], A](mf: A \/ MapFunc[T, TCoMF[T, A]]): CoMFR[T, A] =
+    makeCoMF[T, A, TCoMF[T, A]](mf)
+
   /** Returns a List that maps element-by-element to a MapFunc array. If we
```
Comment (paulp) [inline connector/src/main/scala/quasar/qscript/MapFunc.scala]: And then I might call these methods `coMapFunc` and `coMapFuncR`.

File: connector/src/main/scala/quasar/qscript/MapFunc.scala (hunk 1 above)
Comment (alissapajer) [inline connector/src/main/scala/quasar/qscript/MapFunc.scala]: Those names work for me. What matters to me is that I can easily easily see that the `CoEnv[A, MapFunc[T, ?], T[CoEnv[A, MapFunc[T, ?], ?]]]` in one place is the same as the `CoEnv[A, MapFunc[T, ?], T[CoEnv[A, MapFunc[T, ?], ?]]]` in some other place, and doesn't vary in some slight way.

Comment (paulp) [inline connector/src/main/scala/quasar/qscript/MapFunc.scala]: Right, that's what I mean by common subexpression elimination (where subexpression can be the whole expression.)

**Comment by alissapajer** [comment]
@paulp changes pushed

**Comment by paulp** [comment]
Looks good if travis approves.

**Pull request merged**